src/utils/github.rs

```diff
@@ -14,22 +14,19 @@ struct Commit {
     sha: String,
 }
 
-pub async fn get_latest_commit_hash(org: &str, repo: &str) -> Result<String, GithubError> {
-    let github_api_url = format!("{}/repos/{}/{}/commits", GITHUB_API_BASE_URL, org, repo);
+pub async fn get_latest_commit_hash(org: &str, repo: &str, branch: &str) -> Result<String, GithubError> {
+    let github_api_url = format!("{}/repos/{}/{}/commits/{}", GITHUB_API_BASE_URL, org, repo, branch);
 
     let client = Client::new();
     let response = client.get(github_api_url).header("User-Agent", "reqwest").send().await;
 
-    return match response {
-        Ok(response) => match response.json::<Vec<Commit>>().await {
-            Ok(commits) => match commits.first() {
-                Some(latest_commit) => Ok(latest_commit.sha.clone()),
-                None => Err(GithubError::NoCommitsFound),
-            },
+    match response {
+        Ok(response) => match response.json::<Commit>().await {
+            Ok(latest_commit) => Ok(latest_commit.sha.clone()),
```

Comment:
we don't need to call `first()`?

Reply:
no, when we add the branch to the api it automatically gives the latest commit